x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/upgrade_review_algorithms.md

```diff
@@ -179,8 +217,42 @@ Examples:
 | esql_query         | esql_query  | {query: "FROM query WHERE true", language: "esql"}                                   | {query: "FROM query WHERE false", language: "esql"}                                   | {query: "FROM query WHERE false", language: "esql"}                                   | {query: "FROM query WHERE false", language: "esql"}                                   |
 ```
 
+#### **Scenario: `ABB` - Rule field is rule `type`**
```

Comment:
@dplumlee  The same in this PR for not possible and not possible cases:

https://github.com/elastic/kibana/pull/193369#discussion_r1777645617

Maybe @pborgonovi has an opinion on this: some of the scenarios are not possible in our system (see comment above), but technically we can create an outcome for them. What should be the way forward from a testing perspective? Should we write in the test plan that the scenarios are not possible? Should they have test cases anyways (even though the mock data that needs to be generated in order to create those scenarios is technically not possible to achieve in the normal flow of the application)?

Reply:
Hi @dplumlee and @jpdjere 

For edge scenarios that are not possible within the current flow of our application but can be tested using mock data, I recommend we still include them in our test plan.

We should also provide an explanation of why these scenarios are not achievable in the normal flow, outlining any relevant technical limitations or system constraints. This will help manage stakeholder expectations and ensure everyone is aware of the system's current capabilities. Including these scenarios can also help future-proof the system and ensure it doesn’t break if they arise due to future changes. In the corresponding test cases, we will indicate that the tests depend on artificially generated data.

Let me know if that makes sense.